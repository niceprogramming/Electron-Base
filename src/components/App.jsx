import React from 'react';
import {hot} from 'react-hot-loader';
import styled from 'styled-components';

const Title = styled.h1`
  font-size: 1.5em;
  text-align: center;
  color: palevioletred;
`;

const Wrapper = styled.section`
  background: papayawhip;
  padding: 4em;
`;

const App = () => {
  return(
    <Wrapper>
      <Title>
        Hello World, this is my first styled component!
      </Title>
  </Wrapper>
  );
};

export default hot(module)(App);

