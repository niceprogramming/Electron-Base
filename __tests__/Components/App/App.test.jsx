import React from 'react'
import {render} from 'react-testing-library'
import App from '../../../src/components/App'


test('Check if App DOM is rendered properly', () => {
    const app = render(<App />);

    expect(app).toMatchSnapshot();
  });

