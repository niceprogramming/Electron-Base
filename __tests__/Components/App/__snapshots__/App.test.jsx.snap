// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if App DOM is rendered properly 1`] = `
Object {
  "container": <div>
    <section
      class="App__Wrapper-s1mvb1l6-1 WUJyW"
    >
      <h1
        class="App__Title-s1mvb1l6-0 lkKCXU"
      >
        Hello World, this is my first styled component!
      </h1>
    </section>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
